[bug]: make_xml_documentation tool causes make to stall when using LOCAL_MOD_SUBDIRS

### Severity

Minor

### Versions

18.9-cert4, git master (e0ed0db41f32a47b49a655c737b93f66b292cb04)

### Components/Modules

make, documentation

### Operating Environment

Platform 1: Alma Linux 8 running in a Docker container

Platform 2: Ubuntu 22.04 on an old Intel NUC
5.19.0-38-generic #39~22.04.1-Ubuntu SMP PREEMPT_DYNAMIC Fri Mar 17 21:16:15 UTC 2 x86_64 x86_64 x86_64 GNU/Linux

### Frequency of Occurrence

Constant

### Issue Description

Using LOCAL_MOD_SUBDIRS to set a path for building external modules currently breaks the main make job. Note this is present in 18.9-cert4 as well as master at the time of writing this issue. It was *not* present in 18.9-cert1.

Reproduction steps are very simple:

```
$ mkdir /home/<user>/fake_mod_dir
$ cd /home/<user>/<asterisk_source_dir>
$ export LOCAL_MOD_SUBDIRS=/home/<user>/fake_mod_dir
$ ./configure
$ make
```

A cursory investigation leads me to believe this is caused by documentation generation provided by build_tools/make_xml_documentation -- Investigating the process tree showed that it was the second deepest child process of make and I was able to reproduce the freeze simply by running the following:

```
/bin/sh build_tools/make_xml_documentation --command=print_dependencies --source-tree=. --mod-subdirs="channels pbx apps codecs formats cdr cel ${LOCAL_MOD_SUBDIRS}"
```

Prior to submitting this ticket, I confirmed that setting LOCAL_MOD_SUBDIRS does *not* cause a similar stall when building off a clean asterisk-18.9-cert1 from https://downloads.asterisk.org/pub/telephony/certified-asterisk/releases/asterisk-certified-18.9-cert1.tar.gz with an empty local_mod_subdir

Also I confirmed that this issue *is* reproduced when using a similar clean build off of https://downloads.asterisk.org/pub/telephony/certified-asterisk/releases/asterisk-certified-18.9-cert4.tar.gz

### Relevant log output

```shell
Example output:


$ make
/usr/bin/find: ‘.//home/asterisk/fake_mod_dir’: No such file or directory
```

At this line it's just frozen and never returns

/bin/sh build_tools/make_xml_documentation --command=print_dependencies --source-tree=. --mod-subdirs="channels pbx apps codecs formats cdr cel ${LOCAL_MOD_SUBDIRS}"

yielded the following:

```
$ /bin/sh build_tools/make_xml_documentation --command=print_dependencies --source-tree=. --mod-subdirs="channels pbx apps codecs formats cdr cel ${LOCAL_MOD_SUBDIRS}"
./channels/chan_dahdi.c
./channels/sig_pri.c
./channels/chan_iax2.c
./channels/chan_motif.c
./channels/pjsip/dialplan_functions.c
./pbx/pbx_ael.c
./pbx/pbx_dundi.c
./pbx/pbx_config.c
./apps/app_playback.c
./apps/app_sf.c
./apps/app_userevent.c
./apps/app_saycounted.c
./apps/app_queue.c
./apps/app_externalivr.c
./apps/app_waitforsilence.c
./apps/app_skel.c
./apps/app_ivrdemo.c
./apps/app_sla.c
./apps/app_readexten.c
./apps/app_signal.c
./apps/app_broadcast.c
./apps/app_talkdetect.c
./apps/app_sayunixtime.c
./apps/app_mp3.c
./apps/app_controlplayback.c
./apps/app_getcpeid.c
./apps/app_chanisavail.c
./apps/app_dumpchan.c
./apps/app_stream_echo.c
./apps/app_directory.c
./apps/app_authenticate.c
./apps/app_waitforcond.c
./apps/app_stasis.c
./apps/app_sendtext.c
./apps/app_forkcdr.c
./apps/app_meetme.c
./apps/app_speech_utils.c
./apps/app_bridgewait.c
./apps/app_waitforring.c
./apps/app_minivm.c
./apps/app_channelredirect.c
./apps/app_blind_transfer.c
./apps/app_reload.c
./apps/app_celgenuserevent.c
./apps/app_statsd.c
./apps/app_zapateller.c
./apps/app_dial.c
./apps/app_system.c
./apps/app_festival.c
./apps/app_record.c
./apps/app_voicemail_odbc.c
./apps/app_cdr.c
./apps/app_echo.c
./apps/app_stack.c
./apps/app_directed_pickup.c
./apps/app_sms.c
./apps/app_playtones.c
./apps/app_dictate.c
./apps/app_read.c
./apps/app_bridgeaddchan.c
./apps/app_amd.c
./apps/app_if.c
./apps/app_test.c
./apps/app_disa.c
./apps/app_flash.c
./apps/app_chanspy.c
./apps/app_while.c
./apps/app_voicemail_imap.c
./apps/app_confbridge.c
./apps/confbridge/confbridge_manager.c
./apps/confbridge/conf_config_parser.c
./apps/app_db.c
./apps/app_audiosocket.c
./apps/app_mf.c
./apps/app_dtmfstore.c
./apps/app_adsiprog.c
./apps/app_page.c
./apps/app_milliwatt.c
./apps/app_followme.c
./apps/app_voicemail.c
./apps/app_jack.c
./apps/app_senddtmf.c
./apps/app_verbose.c
./apps/app_agent_pool.c
./apps/app_transfer.c
./apps/app_attended_transfer.c
./apps/app_waituntil.c
./apps/app_privacy.c
./apps/app_exec.c
./apps/app_alarmreceiver.c
./apps/app_originate.c
./apps/app_softhangup.c
./apps/app_morsecode.c
./apps/app_mixmonitor.c
./cdr/cdr_manager.c
./cel/cel_manager.c
find: ‘.//home/asterisk/fake_mod_dir’: No such file or directory
```

Where it was once again frozen.
```


### Asterisk Issue Guidelines

- [X] Yes, I have read the Asterisk Issue Guidelines

Narrowed it down for you guys a little more.

build_tools/make_xml_documentation line 152

`if [ "${command}" = "print_dependencies" ] ; then
        for subdir in ${mod_subdirs} ; do
                subpath="${source_tree}/${subdir}"
                # We WANT word splitting in the following line.
                # shellcheck disable=SC2046
                ${GREP} -l -E '(language="en_US"|appdocsxml.dtd)' $(${FIND} "${subpath}" -name '*.c' -or -name '*.cc' -or -name '*.xml') || :
        done
        exit
fi`

Freeze is in the grep line here. It chokes specifically because we're using an absolute path instead of a path relative to the asterisk source directory.

--- a/build_tools/make_xml_documentation
+++ b/build_tools/make_xml_documentation
@@ -137,10 +137,12 @@ fi
 
 if [ "${command}" = "print_dependencies" ] ; then
 	for subdir in ${mod_subdirs} ; do
-		subpath="${source_tree}/${subdir}"
-		# We WANT word splitting in the following line.
-		# shellcheck disable=SC2046
-		${GREP} -l -E '(language="en_US"|appdocsxml.dtd)' $(${FIND} "${subpath}" -name '*.c' -or -name '*.cc' -or -name '*.xml') || :
+		case "$subdir" in
+			/*) subpath="$subdir" ;;
+			*)  subpath="$source_tree/$subdir" ;;
+		esac
+		${FIND} "${subpath}" \( -name '*.c' -o -name '*.cc' -o -name '*.xml' \) \
+			-exec ${GREP} -l -E '(language="en_US"|appdocsxml.dtd)' '{}' \;
 	done
 	exit
 fi

